src/components/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+import { nanoid } from "nanoid";
+import useBackground from "../../hooks/useBackground";
+import useResponsivePropsCSS from "../../hooks/useResponsivePropsCSS";
+import { responsiveSize } from "../../utils/css";
+
+import Field from "../internal/Field";
+import { InternalTextareaProps } from "./types";
+import { useMergedProps } from "../../hooks/useMergedProps";
+import { defaultTextareaProps } from "./defaultTextareaProps";
+
+const DEFAULT_PROPS = {
+  ...defaultTextareaProps,
+  color: "grey.t05",
+  validate: (value: any, { isEmpty }: any) => {
+    if (isEmpty(value)) {
+      return "Required";
+    }
+
+    return null;
+  },
+} as const;
+
+function Textarea(props: InternalTextareaProps) {
+  const mergedProps = useMergedProps(props, defaultTextareaProps);
+  const {
+    innerRef,
+    value,
+    maxLength,
+    label,
+    placeholder,
+    helpText,
+    disabled,
+    optional,
+    error,
+    onChange,
+    onFocus,
+    onBlur,
+    testId,
+    __internal__focus,
+  } = mergedProps;
+  const [textareaId] = useState(() => `textarea-${nanoid()}`);
+  const [auxId] = useState(() => `textarea-aux-${nanoid()}`);
+
+  const { inputColorMap } = useBackground();
+  const textareaCSS = useResponsivePropsCSS(props, DEFAULT_PROPS, {
+    // @ts-ignore
+    color: (propsAtBreakpoint, theme, bp) => {
+      // @ts-ignore
+      const color = props.color ?? inputColorMap[bp];
+
+      return theme.textarea.getCSS({
+        color,
+        __internal__focus,
+      });
+    },
+    height: responsiveSize("height"),
+  });
+
+  const fieldErrors = Array.isArray(error) ? error : [error];
+  const hasErrors = Array.isArray(error) ? error.length !== 0 : !!error;
+
+  return (
+    <Field
+      optional={optional}
+      disabled={disabled}
+      label={label}
+      labelFor={textareaId}
+      auxId={auxId}
+      helpText={helpText}
+      errors={fieldErrors}
+    >
+      <textarea
+        ref={innerRef}
+        css={textareaCSS}
+        id={textareaId}
+        placeholder={placeholder}
+        maxLength={
+          typeof maxLength === "string" ? Number(maxLength) : maxLength
```

Comment:
We could simply do `maxLength={Number(maxLength)}` here

Reply:
we can't because, `Nubmer(undefined)` is `NaN` which isn't really valid. The point of this was to ensure that if it's undefined or a number we preserve that and only really deal with when it's a string.